

192.168.0.54 可以连接, 但是 192.168.0.91 不可以连接, 原因: 0.91 是8.0.18 版本的.
解决办法 
	root@mysqldb 19:47:  [(none)]> ALTER USER 'root'@'192.168.0.%' IDENTIFIED BY '123456abc' PASSWORD EXPIRE NEVER; 
	Query OK, 0 rows affected (0.24 sec)

	root@mysqldb 19:49:  [(none)]> ALTER USER 'root'@'192.168.0.%' IDENTIFIED WITH mysql_native_password BY '123456abc';
	Query OK, 0 rows affected (0.00 sec)



2019-10-28 19:49:35.602  INFO [WrapperSimpleAppMain] (io.mycat.MycatServer.startup(MycatServer.java:516)) - Perform XA recovery log ...
2019-10-28 19:49:35.603  INFO [WrapperSimpleAppMain] (io.mycat.MycatServer.startup(MycatServer.java:520)) - Perform XA recovery log end...
2019-10-28 19:49:35.608 DEBUG [pool-2-thread-1] (io.mycat.backend.mysql.xa.recovery.impl.FileSystemRepository.init(FileSystemRepository.java:59)) - baseDir ./tmlogs/
2019-10-28 19:49:35.608 DEBUG [pool-2-thread-1] (io.mycat.backend.mysql.xa.recovery.impl.FileSystemRepository.init(FileSystemRepository.java:60)) - baseName tmlog
2019-10-28 19:49:35.610  INFO [Timer0] (io.mycat.backend.datasource.PhysicalDatasource.createByIdleLitte(PhysicalDatasource.java:397)) - create connections ,because idle connection not enough ,cur is 0, minCon is 10 for kp04
2019-10-28 19:49:35.625 DEBUG [$_NIOConnector] (io.mycat.net.AbstractConnection.onConnectfinish(AbstractConnection.java:620)) - 连接后台真正完成
2019-10-28 19:49:35.625 DEBUG [$_NIOConnector] (io.mycat.net.AbstractConnection.onConnectfinish(AbstractConnection.java:620)) - 连接后台真正完成
2019-10-28 19:49:35.627  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.mysql.nio.MySQLConnectionAuthenticator.handle(MySQLConnectionAuthenticator.java:91)) - can't connect to mysql server ,errmsg:Client does not support authentication protocol requested by server; consider upgrading MySQL client MySQLConnection@1982483928 [id=21, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26318, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.628  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.mysql.nio.handler.NewConnectionRespHandler.connectionError(NewConnectionRespHandler.java:37)) - MySQLConnection@1982483928 [id=21, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26318, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false] connectionError ConnectionException [code=1251, msg=Client does not support authentication protocol requested by server; consider upgrading MySQL client]
2019-10-28 19:49:35.628  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.mysql.nio.MySQLConnectionAuthenticator.handle(MySQLConnectionAuthenticator.java:91)) - can't connect to mysql server ,errmsg:Client does not support authentication protocol requested by server; consider upgrading MySQL client MySQLConnection@340831369 [id=22, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26319, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.628  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.mysql.nio.handler.NewConnectionRespHandler.connectionError(NewConnectionRespHandler.java:37)) - MySQLConnection@340831369 [id=22, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26319, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false] connectionError ConnectionException [code=1251, msg=Client does not support authentication protocol requested by server; consider upgrading MySQL client]
2019-10-28 19:49:35.628  INFO [$_NIOREACTOR-0-RW] (io.mycat.net.AbstractConnection.close(AbstractConnection.java:520)) - close connection,reason:stream closed ,MySQLConnection@1982483928 [id=21, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26318, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.628  INFO [$_NIOREACTOR-0-RW] (io.mycat.backend.datasource.PhysicalDatasource$1$1.connectionError(PhysicalDatasource.java:508)) - connection connectionError 
2019-10-28 19:49:35.628  INFO [$_NIOREACTOR-0-RW] (io.mycat.net.AbstractConnection.close(AbstractConnection.java:520)) - close connection,reason:stream closed ,MySQLConnection@340831369 [id=22, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26319, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.628  INFO [$_NIOREACTOR-0-RW] (io.mycat.backend.datasource.PhysicalDatasource$1$1.connectionError(PhysicalDatasource.java:508)) - connection connectionError 
2019-10-28 19:49:35.631 DEBUG [Timer0] (io.mycat.sqlengine.SQLJob.connectionAcquired(SQLJob.java:89)) - con query sql:select user() to con:MySQLConnection@938970472 [id=6, lastTime=1572263375631, user=root, schema=db2, old shema=db2, borrowed=true, fromSlaveDB=false, threadId=66, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.54, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.631  INFO [Timer0] (io.mycat.backend.datasource.PhysicalDatasource.getConnection(PhysicalDatasource.java:564)) - no ilde connection in pool 843960821 ,create new connection for kp04 of schema db3 totalConnectionCount: 0 increamentCount: 11
2019-10-28 19:49:35.633 DEBUG [$_NIOConnector] (io.mycat.net.AbstractConnection.onConnectfinish(AbstractConnection.java:620)) - 连接后台真正完成
2019-10-28 19:49:35.634 DEBUG [$_NIOConnector] (io.mycat.net.AbstractConnection.onConnectfinish(AbstractConnection.java:620)) - 连接后台真正完成
2019-10-28 19:49:35.638 DEBUG [$_NIOREACTOR-0-RW] (io.mycat.backend.datasource.PhysicalDatasource.releaseChannel(PhysicalDatasource.java:633)) - release channel MySQLConnection@938970472 [id=6, lastTime=1572263375601, user=root, schema=db2, old shema=db2, borrowed=true, fromSlaveDB=false, threadId=66, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.54, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.638  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.mysql.nio.MySQLConnectionAuthenticator.handle(MySQLConnectionAuthenticator.java:91)) - can't connect to mysql server ,errmsg:Client does not support authentication protocol requested by server; consider upgrading MySQL client MySQLConnection@817491539 [id=23, lastTime=1572263375601, user=root, schema=db4, old shema=db4, borrowed=false, fromSlaveDB=false, threadId=26320, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.638  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.mysql.nio.handler.NewConnectionRespHandler.connectionError(NewConnectionRespHandler.java:37)) - MySQLConnection@817491539 [id=23, lastTime=1572263375601, user=root, schema=db4, old shema=db4, borrowed=false, fromSlaveDB=false, threadId=26320, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false] connectionError ConnectionException [code=1251, msg=Client does not support authentication protocol requested by server; consider upgrading MySQL client]
2019-10-28 19:49:35.638  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.mysql.nio.MySQLConnectionAuthenticator.handle(MySQLConnectionAuthenticator.java:91)) - can't connect to mysql server ,errmsg:Client does not support authentication protocol requested by server; consider upgrading MySQL client MySQLConnection@533648173 [id=24, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26321, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.638  INFO [$_NIOREACTOR-0-RW] (io.mycat.sqlengine.SQLJob.connectionError(SQLJob.java:125)) - can't get connection for sql :select user()

2019-10-28 19:49:35.638  WARN [$_NIOREACTOR-0-RW] (io.mycat.backend.heartbeat.MySQLDetector.onResult(MySQLDetector.java:208)) - heart beat error: kp04/datahost_91 retry=10 tmo=30000
2019-10-28 19:49:35.638 ERROR [$_NIOREACTOR-0-RW] (io.mycat.backend.heartbeat.MySQLHeartbeat.nextDector(MySQLHeartbeat.java:215)) - set Error 1  DBHostConfig [hostName=kp04, url=192.168.0.91:3306]
2019-10-28 19:49:35.638  INFO [$_NIOREACTOR-0-RW] (io.mycat.net.AbstractConnection.close(AbstractConnection.java:520)) - close connection,reason:stream closed ,MySQLConnection@817491539 [id=23, lastTime=1572263375601, user=root, schema=db4, old shema=db4, borrowed=false, fromSlaveDB=false, threadId=26320, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.638  INFO [$_NIOREACTOR-0-RW] (io.mycat.backend.datasource.PhysicalDatasource$1$1.connectionError(PhysicalDatasource.java:508)) - connection connectionError 
2019-10-28 19:49:35.639  INFO [$_NIOREACTOR-0-RW] (io.mycat.net.AbstractConnection.close(AbstractConnection.java:520)) - close connection,reason:stream closed ,MySQLConnection@533648173 [id=24, lastTime=1572263375601, user=root, schema=db3, old shema=db3, borrowed=false, fromSlaveDB=false, threadId=26321, charset=utf8, txIsolation=3, autocommit=true, attachment=null, respHandler=null, host=192.168.0.91, port=3306, statusSync=null, writeQueue=0, modifiedSQLExecuted=false]
2019-10-28 19:49:35.639  INFO [$_NIOREACTOR-0-RW] (io.mycat.backend.datasource.PhysicalDatasource$1$1.connectionError(PhysicalDatasource.java:508)) - connection connectionError 



	

	
